src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenDictionary.cs

```diff
@@ -181,46 +181,66 @@ public static class FrozenDictionary
                 {
                     if (analysis.RightJustifiedSubstring)
                     {
-                        if (analysis.IgnoreCase)
+                        if (analysis.IgnoreCaseForHash)
                         {
                             frozenDictionary = analysis.AllAsciiIfIgnoreCase
                                 ? new OrdinalStringFrozenDictionary_RightJustifiedCaseInsensitiveAsciiSubstring<TValue>(keys, values, stringComparer, analysis.MinimumLength, analysis.MaximumLengthDiff, analysis.HashIndex, analysis.HashCount)
                                 : new OrdinalStringFrozenDictionary_RightJustifiedCaseInsensitiveSubstring<TValue>(keys, values, stringComparer, analysis.MinimumLength, analysis.MaximumLengthDiff, analysis.HashIndex, analysis.HashCount);
                         }
                         else
                         {
-                            frozenDictionary = analysis.HashCount == 1
-                                ? new OrdinalStringFrozenDictionary_RightJustifiedSingleChar<TValue>(keys, values, stringComparer, analysis.MinimumLength, analysis.MaximumLengthDiff, analysis.HashIndex)
-                                : new OrdinalStringFrozenDictionary_RightJustifiedSubstring<TValue>(keys, values, stringComparer, analysis.MinimumLength, analysis.MaximumLengthDiff, analysis.HashIndex, analysis.HashCount);
+                            if (analysis.HashCount == 1)
```

Comment:
Is pattern matching allowed in this library? if so maybe I'd prefer to express as 
```
frozenDictionary = analysis switch {
 ...
}
```

Reply:
I don't believe they are explicitly disallowed.

Reply:
<img width="1053" alt="image" src="https://github.com/dotnet/runtime/assets/1958108/4c4bbe5d-dcd3-4a13-bd85-02ced8543f09">

I can push this change if preferred. I don't want to be presumptuous as this is perhaps opinionated, so I will wait for confirmation.